File: Baekjoon/Que/1966.py
# https://www.acmicpc.net/problem/1966

t = int(input())
 
for _ in range(t):
    n, m = map(int,input().split())
    arr = list(map(int,input().split()))
    tmp = [0 for _ in range(n)] 
    tmp[m] = 1 
 
    cnt = 0
    while True:
        if arr[0] == max(arr):
            cnt += 1
 
            if tmp[0] != 1:
                arr.pop(0)
                tmp.pop(0)
            else:
                print(cnt)
                break
        else:
            arr.append(arr.pop(0))
            tmp.append(tmp.pop(0))

File: Baekjoon/Que/1966.cpp
// https://www.acmicpc.net/problem/1966

#include <iostream>
#include <queue>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int T;
    cin >> T;

    while (T--) {
        int N, M;
        cin >> N >> M;

        queue<pair<int, int>> q;
        priority_queue<int> pq;
        for (int i = 0; i < N; i++) {
            int x;
            cin >> x;
            q.push({x, i});
            pq.push(x);
        }

        int cnt = 0;
        while (!q.empty()) {
            int x = q.front().first;
            int idx = q.front().second;
            q.pop();
            if (pq.top() == x) {
                pq.pop();
                cnt++;
                if (idx == M) {
                    cout << cnt << '\n';
                    break;
                }
            } else {
                q.push({x, idx});
            }
        }
    }

    return 0;
}